stdlib/public/Darwin/Accelerate/vDSP_Conversion.swift

```diff
@@ -0,0 +1,796 @@
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func convert<U, V>(_ source: U,
+                                     to destination: inout V)
+        where
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        U.Element == Int16,
+        V.Element == Double {
+            
+            precondition(source.count == destination.count)
+            
+            let n = vDSP_Length(source.count)
+            
+            destination.withUnsafeMutableBufferPointer { dest in
+                source.withUnsafeBufferPointer { src in
+                    vDSP_vflt16D(src.baseAddress!, 1,
+                                 dest.baseAddress!, 1,
+                                 n)
+                }
+            }
+    }
+    
+    /// Converts an array of signed 32-bit integers to single-precision floating-point values.
+    ///
+    /// - Parameter source: Source vector.
+    /// - Parameter destination: Destination vector.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func convert<U, V>(_ source: U,
+                                     to destination: inout V)
+        where
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        U.Element == Int32,
+        V.Element == Float {
+            
+            precondition(source.count == destination.count)
+            
+            let n = vDSP_Length(source.count)
+            
+            destination.withUnsafeMutableBufferPointer { dest in
+                source.withUnsafeBufferPointer { src in
+                    vDSP_vflt32(src.baseAddress!, 1,
+                                dest.baseAddress!, 1,
+                                n)
+                }
+            }
+    }
+    
+    /// Converts an array of signed 32-bit integers to double-precision floating-point values.
+    ///
+    /// - Parameter source: Source vector.
+    /// - Parameter destination: Destination vector.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func convert<U, V>(_ source: U,
+                                     to destination: inout V)
+        where
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        U.Element == Int32,
+        V.Element == Double {
+            
+            precondition(source.count == destination.count)
+            
+            let n = vDSP_Length(source.count)
+            
+            destination.withUnsafeMutableBufferPointer { dest in
+                source.withUnsafeBufferPointer { src in
+                    vDSP_vflt32D(src.baseAddress!, 1,
+                                 dest.baseAddress!, 1,
+                                 n)
+                }
+            }
+    }
+    
+    // MARK: Floating-point to integer conversion
+    
+    public enum RoundingMode {
+        case towardsZero
```

Comment:
FWIW: the `FloatingPointRoundingRule` uses `towardZero` (no 's').

Reply:
fixed (and `towardsNearestInteger` too).